testsupport/space.go

```diff
@@ -187,6 +187,7 @@ func verifyResourcesProvisionedForSpace(t *testing.T, hostAwait *wait.HostAwaiti
 	space, err := hostAwait.WaitForSpace(t, spaceName,
 		wait.UntilSpaceHasTier(tier.Name),
 		wait.UntilSpaceHasLabelWithValue(fmt.Sprintf("toolchain.dev.openshift.com/%s-tier-hash", tier.Name), hash),
+		wait.UntilSpaceHasLabelWithValue("appstudio.redhat.com/workspace_name", tier.Name),
```

Comment:
I guess you should check this label on the namespace resources not the `Space`. 

If you have to check this label you need to add a check in the block of code starting from line 217:

```
	if tier.Name == "appstudio" {
		// checks that namespace exists and has the expected label(s)
		ns, err := targetCluster.WaitForNamespace(t, space.Name, tier.Spec.Namespaces[0].TemplateRef, space.Spec.TierName, wait.UntilNamespaceIsActive())
		require.NoError(t, err)
		require.Contains(t, ns.Labels, toolchainv1alpha1.WorkspaceLabelKey)
                 require.Contains(t, ns.Labels, "appstudio.redhat.com/workspace_name")
		assert.Equal(t, space.Name, ns.Labels[toolchainv1alpha1.WorkspaceLabelKey])
	}
``` 

plus maybe add a similar check for the `appstudio-env` tier if required. 

Also the value of your label is not `tier.Name` but `space.Name` I guess.

Reply:
updated the `appstudio` tier check. I could not find anything on `appstudio-env` tier check in space.go. So added checks on the checks.go. should I add a check in the space.go as well for `appstudio-env` tier? how do you think?